lib/controller.js

```diff
@@ -460,150 +460,121 @@ controller.connectToNetwork = function(opts) {
-          return controller.updateTesselWithVersion(opts.force, tessel, currentVersion, requestedVersion)
-            .then(resolve, reject);
-        } else {
-          // If they have requested the latest firmware
-          var verifiedVersion;
-          if (requestedVersion === 'latest') {
-            // Figure out what the version is
-            verifiedVersion = updates.fetchBuildList()
-              .then(function(builds) {
-                // Make sure we actually have builds
-                if (Object.keys(builds).length === 0) {
-                  return reject('Could not detect any builds on server.');
-                }
-                // Sort the builds by version
-                var buildArray = updates.buildDictToSortedArray(builds);
-
-                // Return the newest one
-                return buildArray[0].version;
-              });
+          // If it's not connected via USB, we can't update it
+          if (!tessel.usbConnection) {
+            return reject('Must have Tessel connected over USB to complete update. Aborting update.');
           }
-          // If they requested a specific version
-          else {
-            // Ensure that it exists by fetching the builds
-            verifiedVersion = updates.fetchSHAForBuildVersion(requestedVersion)
-              // If that function resolves, the sha exists so the version exists
-              .then(function() {
-                return requestedVersion;
-              })
-              // The version does not exist
-              .catch(reject);
+
+          // // If this Tessel isn't connected to the LAN
+          if (!tessel.lanConnection || !tessel.lanConnection.authorized) {
+            // Warn the user that it's going to take a while
+            logs.warn('Warning: Authorized LAN connection not detected for this Tessel. For a faster update, cancel and connect Tessel to LAN with the `tessel wifi` command before update.');
           }
-          // Once we have our verified version
-          verifiedVersion.then(function(requestedVersion) {
+
+          // If the update is forced or this version was requested
+          if (opts.version) {
+            // Fetch the requested version
+            return controller.updateTesselWithVersion(opts.force, tessel, currentVersion, build)
+              .then(resolve, reject);
+          } else {
+            // If they have requested the latest firmware
+            var verifiedVersion;
+            if (version === 'latest') {
+              build = builds[builds.length - 1];
+              version = verifiedVersion = build.version;
+            }
+            // If they requested a specific version
+            else {
+              build = updates.findBuild(builds, 'version', version);
+              if (build) {
+                version = build.version;
+                verifiedVersion = version;
+              }
+            }
+
+            if (!verifiedVersion) {
+              return reject();
```

Comment:
"The requested build was not found. Please see the available builds with `tessel update -l`."